IRLS/GetModel_global.ecl

```diff
@@ -101,135 +173,328 @@ EXPORT DATASET(Layout_Model)
                  TRANSFORM(wi_info,
-                           SELF.orig_col:=RIGHT.Y,
+                           SELF.orig_col:=RIGHT.number,
                            SELF:=LEFT),
                  LOOKUP, FEW);
-  X_rep := JOIN(X_in, map_wi, LEFT.wi_id=RIGHT.orig_wi,
-               TRANSFORM(Cell, SELF.wi_id := RIGHT.wi, SELF := LEFT),
-               LOOKUP, MANY, FEW);
-  Y_rep := JOIN(Y_in, map_wi,
-               LEFT.wi_id=RIGHT.orig_wi AND LEFT.Y=RIGHT.col,
-               TRANSFORM(Cell,SELF.wi_id:=RIGHT.wi,SELF.Y:=1,SELF:=LEFT),
-               LOOKUP, MANY, FEW);
+  wi_info assign_addr(wi_info prev, wi_info curr) := TRANSFORM
+    SELF.frst_addr := IF(prev.frst_addr=0, 1, prev.frst_addr+prev.parts);
+    SELF := curr;
+  END;
+  map_wi := ITERATE(w_dep, assign_addr(LEFT, RIGHT));
+  // now to extend independent and dependent records
+  Ext_Nfld dummy_end(wi_info inf, BOOLEAN isDep, UNSIGNED c) := TRANSFORM
+    this_col := ((c-1) DIV inf.parts) + 1;
+    part := (c-1) % inf.parts;
+    SELF.part := part;
+    SELF.addr := part + inf.frst_addr;
+    SELF.mat_rows := IF(isDep, inf.dep_rows, inf.ind_rows);
+    SELF.mat_cols := IF(isDep, 1, inf.dims);
+    SELF.part_rows:= MIN(inf.num_rows, inf.obs - inf.num_rows*(part));
+    SELF.wi := inf.wi;
+    SELF.frst := inf.frst_addr;
+    SELF.parts := inf.parts;
+    SELF.num_rows := inf.num_rows;
+    SELF.number := this_col;
+    SELF.obs := inf.obs;
+    SELF.dims := inf.dims;
+    SELF.inserts := 0;
+    SELF.id := MIN((part+1)*inf.num_rows, inf.obs);
+    SELF.value := 0;
+    SELF.dropMe := TRUE;
+  END;
+  dep_ends := NORMALIZE(map_wi, LEFT.parts, dummy_end(LEFT, TRUE, COUNTER));
+  ind_ends := NORMALIZE(map_wi, LEFT.parts*(LEFT.dims-1),
+                        dummy_end(LEFT, FALSE, COUNTER));
+  // replicate for multiple dep cols and partition
+  Ext_NFld mark_addr(NumericField nf, wi_info inf, BOOLEAN isDep) := TRANSFORM
+    part := ((nf.id-1) DIV inf.num_rows);
+    SELF.part := part;
+    SELF.addr :=  part + inf.frst_addr;
+    SELF.mat_rows := IF(isDep, inf.dep_rows, inf.ind_rows);
+    SELF.mat_cols := IF(isDep, 1, inf.dims);
+    SELF.part_rows:= MIN(inf.num_rows, inf.obs - inf.num_rows*(part));
+    SELF.wi := inf.wi;
+    SELF.frst := inf.frst_addr;
+    SELF.parts := inf.parts;
+    SELF.num_rows := inf.num_rows;
+    SELF.number := IF(isDep, 1, nf.number);
+    SELF.obs := inf.obs;
+    SELF.dims := inf.dims;
+    SELF.inserts := 0;
+    SELF.dropMe := FALSE;
+    SELF := nf;
+  END;
+  repl_ind := JOIN(ind_screened, map_wi, LEFT.wi=RIGHT.orig_wi,
+                   mark_addr(LEFT, RIGHT, FALSE), LOOKUP, MANY);
+  dist_ind := DISTRIBUTE(repl_ind+ind_ends, addr);
+  sorted_ind := SORT(dist_ind, wi, part, number, id, dropMe, LOCAL);
+  grpd_ind := GROUP(sorted_ind, wi, part, number, LOCAL);
+  // add in missing zero entries
+  Ext_NFld find_inserts(Ext_NFld prev, Ext_NFld curr, BOOLEAN isDep) := TRANSFORM
```

Comment:
Yes, but it may not have been a good idea.  At one point I was convinced that I would want to do something slightly different for dependent variables versus the independent variables.  In most of the transforms I do have some slight difference.  Do you think it would be better for the reader if the parameter were removed?

In this particular case, I think the objective would be for you to comment upon odd or possibly ineffective ECL constructs.  I also wanted you to specifically look at the loop and how the data is provided to all of the nodes.